standard-controller/src/main/java/io/enmasse/controller/standard/RouterStatusCache.java

```diff
@@ -54,20 +56,25 @@
 
     void checkRouterStatus() {
         RouterStatusCollector routerStatusCollector = new RouterStatusCollector(routerManagement, checkRouterLinks);
-        List<RouterStatus> routerStatusList = new ArrayList<>();
         List<Pod> routers = kubernetes.listRouters().stream()
                 .filter(Readiness::isPodReady)
                 .collect(Collectors.toList());
 
         log.info("Collecting status from {} routers", routers.size());
+
+        ExecutorCompletionService<RouterStatus> service = new ExecutorCompletionService<>(ForkJoinPool.commonPool());
         for (Pod router : routers) {
+            service.submit(() -> routerStatusCollector.collect(router));
+        }
+        List<RouterStatus> routerStatusList = new ArrayList<>(routers.size());
+        for (int i = 0; i < routers.size(); i++) {
             try {
-                RouterStatus routerStatus = routerStatusCollector.collect(router);
-                if (routerStatus != null) {
-                    routerStatusList.add(routerStatus);
+                RouterStatus status = service.take().get();
+                if (status != null) {
+                    routerStatusList.add(status);
                 }
             } catch (Exception e) {
-                log.info("Error requesting router status from {}. Ignoring", router.getMetadata().getName(), e);
```

Comment:
@k-wall I'm fine with using completion service instead, but I wanted to keep the router id,  as it would give a hint as to which router is not responding or giving a bad response

Reply:
I think the exception thrown by io.enmasse.amqp.RouterManagement#query have the details of the router which is giving the bad response/failing to respond, so I think we'll have the details.